doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,377 @@
+module, we'll create a new array, this time masking the invalid values:
+
+.. ipython:: python
+
+   from numpy import ma
+   nbcases_ma = ma.masked_values(nbcases, -1)
+
+If we look at the ``nbcases_ma`` masked array, this is what we have:
+
+.. ipython:: python
+
+   nbcases_ma
+
+We can see that this is a different kind of array. As mentioned in the
+introduction, it has three attributes (``data``, ``mask`` and ``fill_value``).
+Keep in ming that the ``mask`` attribute has a ``True`` value for elements
+corresponding to **invalid** data (represented by two dashes in the ``data``
+attribute).
+
+.. note::
+   
+   Adding ``-1`` to missing data is not a problem with :func:`numpy.genfromtxt`;
+   in this particular case, substituting the missing value with ``0`` might have
+   been fine, but we'll see later that this is far from a general solution.
+   Also, it is possible to call the :func:`numpy.genfromtxt` function using the
+   ``usemask`` parameter. If ``usemask=True``, :func:`numpy.genfromtxt`
+   automatically returns a masked array.
+
+Let's try and see what the data looks like excluding the first row
+(data from the Hubei province in China) so we can look at the missing data more
+closely:
+
+.. ipython:: python
+
+    plt.plot(dates, nbcases_ma[1:].T, '--');
+    plt.xticks(selected_dates, dates[selected_dates]);
+    @savefig plot_covid_2.png
+    plt.title("COVID-19 cumulative cases from Jan 21 to Feb 3 2020");
+    
+Now that our data has been masked, let's try summing up all the cases in China:
+
+.. ipython:: python
+	     
+    china_masked = nbcases_ma[np.where(locations[:, 1] == 'China')].sum(axis=0)
+    china_masked
+
+Note that ``china_masked`` is a masked array, so it has a different data
+structure than a regular NumPy array. Now, we can access its data directly by
+using the ``.data`` attribute:
+
+.. ipython:: python
+	     
+    china_total = china_masked.data
+    china_total
+
+That is better: no more negative values. However, we can still see that for some
+days, the cumulative number of cases seems to go down (from 835 to 10, for
+example), which does not agree with the definition of "cumulative data". If we
+look more closely at the data, we can see that in the period where there was
+missing data in mainland China, there was valid data for Hong Kong, Taiwan,
+Macau and "Unspecified" regions of China. Maybe we can remove those from the
+total sum of cases in China, to get a better understanding of the data.
+
+First, we'll identify the indices of locations in China:
+
+.. ipython:: python
+
+    indices_china = np.where((locations[:, 1] == 'China') & 
+                             (locations[:, 0] != 'Hong Kong') & 
+                             (locations[:, 0] != 'Taiwan') &
+                             (locations[:, 0] != 'Macau') &
+                             (locations[:, 0] != 'Unspecified*'))
+
+Now,
+
+.. ipython:: python
+
+    indices_china
```

Comment:
Possibly clearer to use:
```suggestion
    china_mask = (
        (locations[:, 1] == 'China') & 
        (locations[:, 0] != 'Hong Kong') & 
        (locations[:, 0] != 'Taiwan') &
        (locations[:, 0] != 'Macau') &
        (locations[:, 0] != 'Unspecified*'))

Now,

.. ipython:: python

    china_mask.nonzero()
```

`np.where` seems to be something that matlab users reach for a lot, but in idiomatic numpy it's frequently unecessary.

Reply:
Nice to know, thanks!